File: Leetcode/169.py
class Solution:
    def majorityElement(self, nums: list[int]) -> int:
        candidate = None
        count = 0
        for num in nums:
            if count == 0:
                candidate = num
                count = 1
            elif candidate == num:
                count += 1
            else:
                count -= 1
        return candidate


nums = [6, 5, 5]
r = Solution().majorityElement(nums)
print(r)

File: Leetcode/169..js
var majorityElement = function (nums) {
  let mElement = NaN;
  let count = 0;
  for (let i = 0; i < nums.length; i++) {
    const element = nums[i];
    if (count == 0) {
      mElement = element;
    }
    count += mElement == element ? 1 : -1
    
    
  }

  return mElement;


};
const nums = [3, 2, 3];
console.log(majorityElement(nums));